pkgs/development/libraries/forge/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig
+, arrayfire, expat, fontconfig, freeimage, freetype, boost
+, mesa, libGLU_combined, glfw3, SDL2, cudatoolkit
+}:
+
+let
+  version = "1.0.4";
+in stdenv.mkDerivation {
+  pname = "forge";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "arrayfire";
+    repo = "forge";
+    rev = "ref/tags/v${version}";
+    sha256 = "00pmky6kccd7pwi8sma79qpmzr2f9pbn6gym3gyqm64yckw6m484";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkgconfig
+  ];
+
+  buildInputs = [
+    expat
+    fontconfig
+    freetype
+    boost.out
+    boost.dev
+    freeimage
+    mesa
+    libGLU_combined
+    glfw3
+    SDL2
+    cudatoolkit
+    arrayfire
+  ];
+
+  meta = with stdenv.lib; {
+    description = "An OpenGL interop library that can be used with ArrayFire or any other application using CUDA or OpenCL compute backend.";
```

Comment:
(nitpick)
Manual recommends to remove any trailing dot in the short `description`